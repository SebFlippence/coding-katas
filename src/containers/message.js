import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

export class Message extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      display: props.message,
    };
  }

  render() {
    return (
      <Fragment>
        <dt>
          {this.props.avatar && (
            <img
              src={this.props.avatar}
              alt={`Avatar of ${this.props.email}`}
              loading="lazy"
              width="100"
              height="100"
            />
          )}
          <br />
          {new Date(this.props.timestamp).toLocaleString(undefined, { dateStyle: 'full', timeStyle: 'short' })}
        </dt>
        <dd
          onMouseEnter={() => this.setState({ display: this.props.email })}
          onMouseLeave={() => this.setState({ display: this.props.message })}
        >
          {this.state.display}
        </dd>
      </Fragment>
    );
  }
}

Message.propTypes = {
  timestamp: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  avatar: PropTypes.string,
  message: PropTypes.string.isRequired,
};
