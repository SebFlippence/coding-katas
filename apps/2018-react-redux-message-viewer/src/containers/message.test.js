import React from 'react';
import { shallow } from 'enzyme';

import { Message } from './message';

const props = {
  timestamp: '2016-11-09T05:04:58Z',
  email: 'mbradley0@google.it',
  avatar: 'http://dummyimage.com/100x100.png/5fa2dd/ffffff',
  message: 'Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.',
};

it('renders correctly', () => {
  const wrapper = shallow(<Message {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it('handles JavaScript hover', () => {
  const wrapper = shallow(<Message {...props} />);
  wrapper.find('dd').prop('onMouseEnter')();
  expect(wrapper).toMatchSnapshot();
  wrapper.find('dd').prop('onMouseLeave')();
  expect(wrapper).toMatchSnapshot();
});
