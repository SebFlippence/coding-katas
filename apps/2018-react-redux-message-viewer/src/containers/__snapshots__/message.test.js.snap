// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles JavaScript hover 1`] = `
<Fragment>
  <dt>
    <img
      alt="Avatar of mbradley0@google.it"
      height="100"
      loading="lazy"
      src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
      width="100"
    />
    <br />
    11/9/2016, 5:04:58 AM
  </dt>
  <dd
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    mbradley0@google.it
  </dd>
</Fragment>
`;

exports[`handles JavaScript hover 2`] = `
<Fragment>
  <dt>
    <img
      alt="Avatar of mbradley0@google.it"
      height="100"
      loading="lazy"
      src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
      width="100"
    />
    <br />
    11/9/2016, 5:04:58 AM
  </dt>
  <dd
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
  </dd>
</Fragment>
`;

exports[`renders correctly 1`] = `
<Fragment>
  <dt>
    <img
      alt="Avatar of mbradley0@google.it"
      height="100"
      loading="lazy"
      src="http://dummyimage.com/100x100.png/5fa2dd/ffffff"
      width="100"
    />
    <br />
    11/9/2016, 5:04:58 AM
  </dt>
  <dd
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.
  </dd>
</Fragment>
`;
